x/programs/simulator/ffi/ffi.go

```diff
@@ -154,4 +154,40 @@ func newCallProgramResponse(result []byte, fuel uint64, err error) C.CallProgram
 	}
 }
 
+//export GetBalance
+func GetBalance(db *C.Mutable, address *C.Address) C.uint64_t {
+	if db == nil || address == nil {
+		panic(ErrInvalidParam)
+	}
+
+	state := simState.NewSimulatorState(unsafe.Pointer(db))
+	pState := simState.NewProgramStateManager(state)
+	account := C.GoBytes(unsafe.Pointer(address), codec.AddressLen)     //nolint:all
+
+	balance, err := pState.GetBalance(SimContext, codec.Address(account))
+	if err != nil {
+		panic(err)
```

Comment:
i think panicing is fine here, saves extra overhead by not passing error messages go <-> rust. plus if it errors, then something strange is wrong anyways